### PR TITLE
mutter: update to 3.32.0

### PR DESCRIPTION
**Pull request by Oreo639**
fixes void-linux/void-packages#9714

### CODE REVIEW

**Comment by Cogitri** [comment]
Is this patch upstream?

**Comment by Oreo639** [comment]
Yes, however I am working on compiling the latest instead of just adding the patch.
They switched the build system from gnu-configure to meson (and I am trying to update the template), which is why it is taking me so long.

**Comment by Oreo639** [comment]
The issue is a result of updating gsettings-desktop-schemas without updating mutter or other gnome software to accommodate for the changes, since the name of an option was changed.

**Comment by Cogitri** [comment]
Yes, I'm in the process of updating everything to GNOME 3.32, that might take a little.

**Comment by Cogitri** [comment]
So uhh...feel free to ping me if you need help

**Comment by Oreo639** [comment]
Sure.

**Comment by Oreo639** [comment]
My issue is that I'm getting
```
=> WARNING: mutter-3.32.0_1: libmutter-4.so.0 not found in common/shlibs!
=> WARNING: mutter-3.32.0_1: libmutter-clutter-4.so.0 not found in common/shlibs!
=> WARNING: mutter-3.32.0_1: libmutter-cogl-gles2-4.so.0 not found in common/shlibs!
=> WARNING: mutter-3.32.0_1: libmutter-cogl-4.so.0 not found in common/shlibs!
=> WARNING: mutter-3.32.0_1: libmutter-cogl-path-4.so.0 not found in common/shlibs!
=> WARNING: mutter-3.32.0_1: libmutter-cogl-pango-4.so.0 not found in common/shlibs!
=> WARNING: mutter-3.32.0_1: libmutter-clutter-4.so not found in common/shlibs!
=> WARNING: mutter-3.32.0_1: libmutter-cogl-4.so not found in common/shlibs!
=> WARNING: mutter-3.32.0_1: libmutter-cogl-gles2-4.so not found in common/shlibs!
=> WARNING: mutter-3.32.0_1: libmutter-cogl-pango-4.so not found in common/shlibs!
=> WARNING: mutter-3.32.0_1: libmutter-cogl-path-4.so not found in common/shlibs!
```
and when I try to compile gnome-shell, I get:
```
   SONAME: libmutter-clutter-4.so.0 <-> UNKNOWN PKG PLEASE FIX!
   SONAME: libmutter-cogl-pango-4.so.0 <-> UNKNOWN PKG PLEASE FIX!
   SONAME: libgirepository-1.0.so.1 <-> libgirepository>=1.30_1
   SONAME: libmutter-4.so.0 <-> UNKNOWN PKG PLEASE FIX!
   SONAME: libatk-1.0.so.0 <-> atk>=1.26.0_1
   SONAME: libpango-1.0.so.0 <-> pango>=1.24.0_1
   SONAME: libmutter-cogl-4.so.0 <-> UNKNOWN PKG PLEASE FIX!
```

**Comment by Cogitri** [comment]
Yes, the sonames have been bumped, you have to change that in common/shlibs.

**Comment by Cogitri** [comment]
And also shlib_provides in the mutter template. Am on phone right now, I can send you a diff once I'm home

**Comment by Oreo639** [comment]
> Yes, the sonames have been bumped, you have to change that in common/shlibs.
You're right. I forgot to do that. Thanks.
I'll try updating that when I am available.

**Comment by Cogitri** [comment]
Please apply https://gist.github.com/4e6e82091d3adc3cfd8705f15042f30e and it should work.

**Comment by Cogitri** [comment]
Thanks for notifying me about the GNOME breakage BTW, sorry for breaking it. Let's get this fixed soon :)

File: srcpkgs/mutter/patches/meta-renderer-x11.patch [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
-+++ src/backends/x11/meta-renderer-x11.c	2017-06-05 12:14:08.090473267 -0600
-@@ -79,14 +79,6 @@
-   cogl_renderer_set_custom_winsys (cogl_renderer, get_x11_cogl_winsys_vtable);
+--- src/backends/x11/meta-renderer-x11.c	2019-03-11 16:01:02.000000000 -0700
++++ src/backends/x11/meta-renderer-x11.c	2019-03-11 23:44:53.274968331 -0700
+@@ -85,14 +85,6 @@
    cogl_xlib_renderer_set_foreign_display (cogl_renderer, xdisplay);
+   cogl_xlib_renderer_request_reset_on_video_memory_purge (cogl_renderer, TRUE);
  
 -  /* Set up things so that if the INTEL_swap_event extension is not present,
```
Comment (Oreo639) [inline srcpkgs/mutter/patches/meta-renderer-x11.patch]: Also, just out of curiosity, why is this patched out?

Comment (Cogitri) [reply srcpkgs/mutter/patches/meta-renderer-x11.patch]: I think this was about some weird lags on Intel? Dunno and sadly there's no patch header :/

**Comment by Cogitri** [comment]
Also, currently working on switching on udev and native-backend, that's both required for Wayland.

**Comment by Cogitri** [comment]
Please also apply https://gist.github.com/ac22436de7655de64cc522c788671426

File: srcpkgs/mutter/template [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
-version=3.30.2
-revision=2
-build_style=gnu-configure
+version=3.32.0
+revision=1
 build_helper="gir"
-configure_args="--disable-schemas-compile --disable-static --enable-egl-device
- --enable-remote-desktop"
+build_style=meson
+configure_args="-Degl_device=true -Dudev=false -Dnative_backend=false
```
Comment (Cogitri) [inline srcpkgs/mutter/template]: Please switch udev and native_backend to true

File: srcpkgs/mutter/template [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
+ -Dintrospection=true -Dxwayland_path='/bin/Xwayland'"
```
Comment (Cogitri) [inline srcpkgs/mutter/template]: Please remove the quotes here and make this "/usr/bin/Xwayland"

File: srcpkgs/mutter/template (hunk 2 above)
Comment (Cogitri) [reply srcpkgs/mutter/template]: Please also enable egl_device for Nvidia users

**Comment by Cogitri** [comment]
Screenshot from 2019-03-12 19-27-38

Works now :)

**Comment by Oreo639** [comment]
Tysm!

**Comment by maxice8** [comment]
Can be merged?

**Comment by Cogitri** [comment]
No.

**Comment by maxice8** [comment]
Ack

**Comment by Oreo639** [comment]
Adding them right now.

**Comment by Oreo639** [comment]
I applied the changes, I still need to test compiling gnome-shell with it though.

File: srcpkgs/mutter/template (hunk 2 above)
Comment (Oreo639) [reply srcpkgs/mutter/template]: Isn't egl device is already enabled? But I did enable the other two as requested.

**Comment by Oreo639** [comment]
I though xorg-server contained cvt, but on aarch64 and armv7 non-musl doesn't seem to get this file?

**Comment by Oreo639** [comment]
Hm, that's strange. Looking at the aarch64 package for xorg-server, there is cvt.

**Comment by Oreo639** [comment]
Ok, I fixed it. (I accidently set cvt as a buildtarget dep instead of a host dep)

**Comment by maxice8** [comment]
Ping, mergeability?

**Comment by Cogitri** [comment]
will test again.

**Comment by Cogitri** [comment]
wfm, please merge together with #9748

**Pull request merged**